azurerm/resource_arm_network_watcher_test.go

```diff
@@ -14,7 +14,7 @@ func TestAccAzureRMNetworkWatcher_basic(t *testing.T) {
 	resourceGroup := "azurerm_network_watcher.test"
 	rInt := acctest.RandInt()
 	resource.Test(t, resource.TestCase{
```

Comment:
rather than shoe-horning this into the `testAccPreCheck` method - I think it might be worth overriding/sub-classing `resource.TestCase` to expose a `RunInParallel` field as per the original PR - to match the design of the API? e.g.

```
resource.Test(t, resource.TestCase{
  PreCheck: func() {},
  Providers: testAccProviders,
  RunInParallel: false,
  CheckDestroy: #...
})
```

Reply:
@tombuildsstuff We don't want to change 600+ test cases now, that's one of the biggest concern since the beginning of this PR.

In another [PR](https://github.com/hashicorp/terraform/pull/16807/files), we discussed the pain point of 600+ test cases, and @vancluever suggested me to change testAccPreCheck to inject parallelism logic, since this is the smallest change but achieving the goal in a very short time.

We do have plan to add a middle layer, aztesting, to extend functionalities of resource.Test, especially if Terraform core runtime team don't want to add some features into core. That will take even more time to finish the work, I suggest we finish this PR asap considering it's been pending for almost 2 weeks and we're blocked by it to enable Jenkins CI on our side as monitoring, and at the same time, we start another thread to do the code refactoring.